src/gluonts/zebras/_split_frame.py

```diff
@@ -0,0 +1,387 @@
+            _future=rows_to_columns(pluck("_future"), np.stack),  # type: ignore
+            index=pluck("index"),
+            static=rows_to_columns(pluck("static"), np.stack),  # type: ignore
+            past_length=ref.past_length,
+            future_length=ref.future_length,
+            tdims=ref.tdims,
+            metadata=pluck("metadata"),
+            _pad=pluck("_pad"),
+        )
+
+
+@dataclasses.dataclass
+class BatchSplitFrame:
+    _past: dict
+    _future: dict
+    index: List[Optional[Periods]]
+    static: dict
+    past_length: int
+    future_length: int
+    tdims: dict
+    metadata: List[Optional[dict]]
+    _pad: List[Pad]
+
+    @property
+    def batch_size(self):
+        return len(self.index)
+
+    def __len__(self):
+        return self.past_length + self.future_length
+
+    @property
+    def past(self):
+        return BatchTimeFrame(
+            columns=self._past,
+            static=self.static,
+            length=self.past_length,
+            index=[
+                maybe.map(index, itemgetter(slice(None, self.past_length)))
+                for index in self.index
+            ],
+            tdims=self.tdims,
+            metadata=self.metadata,
+            _pad=self._pad,
+        )
+
+    @property
+    def future(self):
+        return BatchTimeFrame(
+            columns=self._future,
+            static=self.static,
+            length=self.future_length,
+            index=[
+                maybe.map(index, itemgetter(slice(self.past_length, None)))
+                for index in self.index
+            ],
+            tdims=self.tdims,
+            metadata=self.metadata,
+            _pad=self._pad,
+        )
+
+    def items(self):
+        return BatchSplitFrameItems(self)
+
+    def as_dict(self):
+        return SplitFrame.as_dict(self)
+
+
+@dataclasses.dataclass(repr=False)
+class BatchSplitFrameItems:
+    data: BatchSplitFrame
+
+    def __len__(self):
+        return self.data.batch_size
+
+    def __getitem__(self, idx):
+        tdims = self.data.tdims
+
+        if isinstance(idx, int):
+            cls = SplitFrame
+            tdims = valmap(lambda tdim: tdim - 1 if tdim >= 0 else tdim, tdims)
+        else:
+            cls = BatchSplitFrame
+
+        return cls(
+            _past=valmap(itemgetter(idx), self.data._past),
+            _future=valmap(itemgetter(idx), self.data._future),
+            index=self.data.index[idx],
+            static=valmap(itemgetter(idx), self.data.static),
+            tdims=tdims,
+            metadata=self.data.metadata[idx],
+            _pad=self.data._pad[idx],
+            past_length=self.data.past_length,
+            future_length=self.data.future_length,
+        )
+
+
+def split_frame(
+    full=None,
+    *,
+    past=None,
+    future=None,
+    past_length=None,
+    future_length=None,
+    metadata=None,
+    static=None,
+    start=None,
+    freq=None,
+    index=None,
+    tdims=None,
+    default_tdim=-1,
+):
```

Comment:
This appears to be a fundamental entry point for the package, maybe add a docstring?